lib/taskinator/process.rb

```diff
@@ -258,9 +258,6 @@ def start
       end
 
       def task_completed(task)
-        # skip if failed
-        return if failed?
-
```

Comment:
Now that this is removed, the other issue that I encountered was the following:
eg.
concurrent process with 3 tasks.
1 task failes
2 tasks complete

When retrying the failed task, the following is called: `pending = deincr_pending_tasks`
And I've noticed that at this point, `pending` is 2. I have not looked too far into this, but I believe that on retry, it is starting pending at total tasks, not the amount of tasks that are still left to run.
The other 2 tasks won't be retried, so `pending` will not reduce to less than 2
Therefore, `complete!` will never be called.

Reply:
This change will fix that scenario too, since in the original implementation once the process status is set to failed due to one or more failing tasks, it didn't de-increment the pending task count when any subsequent tasks completed successfully.

I'll add a further scenario to the spec to cover this scenario to assert this.

Reply:
@ChristopherRollins I've updated the PR with spec changes to cover pending task counts.
I picked up an issue with sequential tasks not doing the same, so have fixed that too.

Reply:
I've tested locally against my test pack for this, and it is working. Thank you.